src/java/org/apache/cassandra/cql3/restrictions/SimpleRestriction.java

```diff
@@ -229,18 +231,42 @@ private List<ClusteringElements> bindAndGetClusteringElements(QueryOptions optio
         {
             case SINGLE_COLUMN:
             case TOKEN:
-                return bindAndGet(options).stream()
-                                          .map(b ->  ClusteringElements.of(columnsExpression.columnSpecification(), b))
-                                          .collect(Collectors.toList());
+                return bindAndGetSingleTermClusteringElements(options);
             case MULTI_COLUMN:
-                return bindAndGetElements(options).stream()
-                                                  .map(buffers -> ClusteringElements.of(columnsExpression.columns(), buffers))
-                                                  .collect(Collectors.toList());
+                return bindAndGetMultiTermClusteringElements(options);
             default:
                 throw new UnsupportedOperationException();
         }
     }
 
+    private List<ClusteringElements> bindAndGetSingleTermClusteringElements(QueryOptions options)
+    {
+        List<ByteBuffer> values = bindAndGet(options);
+        if (!values.isEmpty()) {
+            List<ClusteringElements> elements = new ArrayList<>();
+            for (ByteBuffer value : values) {
+                ClusteringElements byteBuffers = ClusteringElements.of(columnsExpression.columnSpecification(), value);
+                elements.add(byteBuffers);
+            }
+            return elements;
```

Comment:
@samueldlightfoot just wrap `elements` in `Collections.unmodifiableList(elements)` if you need it to be unmodifiable

Reply:
The stream based approach did not return an immutable list, but given this collection is exposed as part of the public interface I believe this is a good idea.